workflow/controller/operator.go

```diff
@@ -1487,6 +1487,9 @@ func (woc *wfOperationCtx) executeResource(nodeName string, tmpl *wfv1.Template,
 		},
 		Env: execEnvVars,
 	}
+	if woc.controller.Config.MainContainerResources != nil {
+		mainCtr.Resources = *woc.controller.Config.MainContainerResources
```

Comment:
It seems this logic will blow away the resources that the user defines. We need to honor the resources defined in the main container, and only fall back to the controller config if unspecified.

Reply:
but in executeResource, users don't have the opportunity to specify the resource, right? As the mainCtr is created by argo itself. Or I miss anything here?

Reply:
@jessesuen   comments?